File: app/src/main/java/com/test/xyz/demo/domain/common/di/CommonModule.java
package com.test.xyz.demo.domain.common.di;

import android.app.Application;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import okhttp3.Cache;
import okhttp3.OkHttpClient;

@Module
public class CommonModule {

    @Provides
    @Singleton
    OkHttpClient provideOkHttpClient(Application application) {
        int cacheSize = 15 * 1024 * 1024; // 15 MB
        Cache cache = new Cache(application.getCacheDir(), cacheSize);

        //HttpLoggingInterceptor interceptor = new HttpLoggingInterceptor();
        //interceptor.setLevel(HttpLoggingInterceptor.Level.BODY);

        OkHttpClient okHttpClient = new OkHttpClient.Builder()
                .cache(cache)
                //.addInterceptor(interceptor)
                .build();

        return okHttpClient;
    }
}

File: app/src/main/kotlin/com/test/xyz/demo/domain/common/di/CommonModule.kt
package com.test.xyz.demo.domain.common.di

import android.app.Application
import dagger.Module
import dagger.Provides
import okhttp3.Cache
import okhttp3.OkHttpClient
import javax.inject.Singleton

@Module
class CommonModule {

    @Provides
    @Singleton
    internal fun provideOkHttpClient(application: Application): OkHttpClient {
        val cacheSize = 15 * 1024 * 1024 // 15 MB
        val cache = Cache(application.cacheDir, cacheSize.toLong())

        return OkHttpClient.Builder()
                .cache(cache)
                .build()
    }
}
